pkg/smartcontract/rpcbinding/binding.go

```diff
@@ -402,29 +407,53 @@ func Generate(cfg binding.Config) error {
 	var imports = make(map[string]struct{})
 	var ctr ContractTmpl
 
+	var isNEP11D, isNEP11ND, isNEP17, isNEP24 bool
 	// Strip standard methods from NEP-XX packages.
 	for _, std := range cfg.Manifest.SupportedStandards {
-		if std == manifest.NEP11StandardName {
+		switch std {
+		case manifest.NEP11StandardName:
 			imports["github.com/nspcc-dev/neo-go/pkg/rpcclient/nep11"] = struct{}{}
 			if standard.ComplyABI(cfg.Manifest, standard.Nep11Divisible) == nil {
-				mfst.ABI.Methods = dropStdMethods(mfst.ABI.Methods, standard.Nep11Divisible)
-				ctr.IsNep11D = true
+				isNEP11D = true
 			} else if standard.ComplyABI(cfg.Manifest, standard.Nep11NonDivisible) == nil {
-				mfst.ABI.Methods = dropStdMethods(mfst.ABI.Methods, standard.Nep11NonDivisible)
-				ctr.IsNep11ND = true
+				isNEP11ND = true
+			}
+		case manifest.NEP17StandardName:
+			if standard.ComplyABI(cfg.Manifest, standard.Nep17) == nil {
+				imports["github.com/nspcc-dev/neo-go/pkg/rpcclient/nep17"] = struct{}{}
+				isNEP17 = true
+			}
+		case manifest.NEP24StandardName:
+			if standard.ComplyABI(cfg.Manifest, standard.Nep24) == nil {
+				imports["github.com/nspcc-dev/neo-go/pkg/rpcclient/nep24"] = struct{}{}
+				isNEP24 = true
 			}
-			mfst.ABI.Events = dropStdEvents(mfst.ABI.Events, standard.Nep11Base)
-			break // Can't be NEP-17 at the same time.
-		}
-		if std == manifest.NEP17StandardName && standard.ComplyABI(cfg.Manifest, standard.Nep17) == nil {
-			mfst.ABI.Methods = dropStdMethods(mfst.ABI.Methods, standard.Nep17)
-			imports["github.com/nspcc-dev/neo-go/pkg/rpcclient/nep17"] = struct{}{}
-			ctr.IsNep17 = true
-			mfst.ABI.Events = dropStdEvents(mfst.ABI.Events, standard.Nep17)
-			break // Can't be NEP-11 at the same time.
 		}
 	}
 
+	// Now drop methods and events for each applicable standard after the loop
+	if isNEP11D {
+		mfst.ABI.Methods = dropStdMethods(mfst.ABI.Methods, standard.Nep11Divisible)
+		ctr.IsNep11D = true
```

Comment:
OK, I see, you may even go without these extra `isNEP11D, ...` variables. Use `ctr.IsNep11D...` fields directly.